west build error after while "getting started" on ESP32

**Describe the bug**
When following the getting started guide I encounter the error:
```
No such file or directory: LIBGCC_FILE_NAME: ''
Call Stack (most recent call first):
  /home/.../zephyrproject/zephyr/cmake/compiler/gcc/target.cmake:69 (assert_exists)
```
which looks similar to #8977 and #23418.
However the toolchain seems to be right and the build tools changed or differ from the above mentioned issues (see below for my west output). I tried the suggested comments there, but most did not yield more information because the tool paths/setup has changed as it seems.

I hope I did not miss anything obvious here, am greatful for help, and will happily provide more information.

**To Reproduce**
follow the [getting started guide](https://docs.zephyrproject.org/latest/getting_started/index.html)

**Expected behavior**
successful build

**Impact**
does not work

**Screenshots or console output**
```
Including boilerplate (Zephyr base (cached)): /home/.../zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/.../zephyrproject/zephyr/samples/hello_world
-- Zephyr version: 2.2.99 (/home/.../zephyrproject/zephyr)
-- Board: esp32
-- Found west: /home/.../.local/bin/west (found suitable version "0.7.2", minimum required is "0.7.1")
-- Found dtc: /home/.../zephyr-sdk-0.11.2/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found toolchain: zephyr (/home/.../zephyr-sdk-0.11.2)
-- Found BOARD.dts: /home/.../zephyrproject/zephyr/boards/xtensa/esp32/esp32.dts
-- Generated zephyr.dts: /home/.../zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /home/.../zephyrproject/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
Parsing /home/.../zephyrproject/zephyr/Kconfig
Loaded configuration '/home/.../zephyrproject/zephyr/build/zephyr/.config'
No change to configuration in '/home/.../zephyrproject/zephyr/build/zephyr/.config'
No change to Kconfig header in '/home/.../zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
CMake Error at /home/.../zephyrproject/zephyr/cmake/extensions.cmake:1479 (message):
  No such file or directory: LIBGCC_FILE_NAME: ''
Call Stack (most recent call first):
  /home/.../zephyrproject/zephyr/cmake/compiler/gcc/target.cmake:65 (assert_exists)
  /home/.../zephyrproject/zephyr/cmake/target_toolchain.cmake:49 (include)
  /home/.../zephyrproject/zephyr/cmake/app/boilerplate.cmake:516 (include)
  /home/.../zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /home/.../zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  CMakeLists.txt:5 (find_package)
```

**Environment (please complete the following information):**
 - OS: Ubuntu Linux 20.04
 - Toolchain: Zephyr SDK (esp-idf also installed previously on system, just in case this might influence something)
 - Commit SHA 61c6e2b17247d4bfb7e24b5cf2d6c2c292ea2d5c

**Additional context**
Before that I also had the error message 
`ERROR: ld.so: object 'libgtk3-nocsd.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored. `
which I fixed w/ the additon of `export LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libgtk3-nocsd.so.0` to .zephyrrc  as suggested in this [stackoverflow question](https://stackoverflow.com/questions/53825857/error-ld-so-object-libgtk3-nocsd-so-0-from-ld-preload-cannot-be-preloaded)
(is this something common or did I do something wrong? Maybe it could be added to the getting started if it is the first case)

Thanks for the great work :-)

@carlescufi oh, you are absolutely right, I am sorry for that.

I am building for the esp32 w/
```west build -p auto -b esp32 samples/hello_world```

Now I re-read the board page and saw that I missed the board specific setup path (I was just looking for the board name before). Alright, did that now but am greeted with a new error. It finds the zephyr and esp toolchains but still complains that ```Zephyr was unable to find the toolchain.  Is the environment misconfigured?```

Is there a better place to ask questions than here? This is probably not a bug but a problem at my side (although the ```export LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libgtk3-nocsd.so.0``` was still needed to get so far)

Thanks and best regards!

```
Including boilerplate (Zephyr base (cached)): /home/.../zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/.../zephyrproject/zephyr/samples/hello_world
-- Zephyr version: 2.2.99 (/home/.../zephyrproject/zephyr)
-- Board: esp32
-- Found toolchain: zephyr (/home/.../zephyr-sdk-0.11.2)
-- Found west: /home/.../.local/bin/west (found suitable version "0.7.2", minimum required is "0.7.1")
-- Found dtc: /home/.../zephyr-sdk-0.11.2/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found toolchain: espressif (/home/.../.espressif/tools/xtensa-esp32-elf/esp-2020r1-8.2.0/xtensa-esp32-elf/xtensa-esp32-elf)
CMake Error at /home/.../zephyrproject/zephyr/cmake/compiler/gcc/generic.cmake:8 (message):
  Zephyr was unable to find the toolchain.  Is the environment misconfigured?

  User-configuration:

  ZEPHYR_TOOLCHAIN_VARIANT: espressif

  Internal variables:

  CROSS_COMPILE:
  /home/.../.espressif/tools/xtensa-esp32-elf/esp-2020r1-8.2.0/xtensa-esp32-elf/xtensa-esp32-elf/bin/xtensa-esp32-elf-


  TOOLCHAIN_HOME:
  /home/.../.espressif/tools/xtensa-esp32-elf/esp-2020r1-8.2.0/xtensa-esp32-elf/xtensa-esp32-elf


Call Stack (most recent call first):
  /home/.../zephyrproject/zephyr/cmake/generic_toolchain.cmake:70 (include)
  /home/.../zephyrproject/zephyr/cmake/app/boilerplate.cmake:502 (include)
  /home/.../zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /home/.../zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /usr/bin/cmake -B/home/.../zephyrproject/zephyr/build -S/home/.../zephyrproject/zephyr/samples/hello_world -GNinja

```

I confirm the problem and the introduction of it in 299a154fddfee098232fef804a5041e82344d527.  I do set `ZEPHYR_BASE` in my environment.  The build trace I get is:

```
tirzah[9]$ rm -rf build && west build -b esp32
-- west build: generating a build system
Including boilerplate (Zephyr base): /mnt/nordic/zp/zephyr/cmake/app/boilerplate.cmake
-- Application: /mnt/nordic/zp/zephyr/samples/hello_world
-- Zephyr version: 2.2.99 (/mnt/nordic/zp/zephyr)
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.2") found components: Interpreter 
-- Board: esp32
-- Found toolchain: zephyr (/usr/local/zephyr-sdk-0.11.3-beta-1-pr-210)
-- Found west: /home/pab/.local/bin/west (found suitable version "0.7.2", minimum required is "0.7.1")
-- Found dtc: /usr/local/zephyr-sdk-0.11.3-beta-1-pr-210/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: /mnt/nordic/zp/zephyr/boards/xtensa/esp32/esp32.dts
-- Generated zephyr.dts: /mnt/nordic/zp/zephyr/samples/hello_world/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /mnt/nordic/zp/zephyr/samples/hello_world/build/zephyr/include/generated/devicetree_unfixed.h
Parsing /mnt/nordic/zp/zephyr/Kconfig
Loaded configuration '/mnt/nordic/zp/zephyr/boards/xtensa/esp32/esp32_defconfig'
Merged configuration '/mnt/nordic/zp/zephyr/samples/hello_world/prj.conf'
Configuration saved to '/mnt/nordic/zp/zephyr/samples/hello_world/build/zephyr/.config'
Kconfig header saved to '/mnt/nordic/zp/zephyr/samples/hello_world/build/zephyr/include/generated/autoconf.h'
CMake Error at /mnt/nordic/zp/zephyr/cmake/extensions.cmake:1479 (message):
  No such file or directory: LIBGCC_FILE_NAME: ''
Call Stack (most recent call first):
  /mnt/nordic/zp/zephyr/cmake/compiler/gcc/target.cmake:65 (assert_exists)
  /mnt/nordic/zp/zephyr/cmake/target_toolchain.cmake:49 (include)
  /mnt/nordic/zp/zephyr/cmake/app/boilerplate.cmake:516 (include)
  /mnt/nordic/zp/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /mnt/nordic/zp/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:35 (include_boilerplate)
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /usr/bin/cmake -B/mnt/nordic/zp/zephyr/samples/hello_world/build -S/mnt/nordic/zp/zephyr/samples/hello_world -GNinja -DBOARD=esp32
```

Still investigating, but just tried to take the commit BEFORE https://github.com/zephyrproject-rtos/zephyr/commit/299a154fddfee098232fef804a5041e82344d527 which is https://github.com/zephyrproject-rtos/zephyr/commit/6789ecbd5e13cc19e69f25314cf486493df9dde3 . 

And I still get: 
```
> git checkout 299a154~1
HEAD is now at 6789ecbd5e usb: samples: Add USB Audio Headset sample
> west update > /dev/null 
HEAD is now at 542b229 DSP: Integrate CMSIS-DSP 1.8.0 (CMSIS 5.7.0)
HEAD is now at 1fe96f0 samv71: patch: Fix GMAC priority queue register offsets
...
> rm -rf build; west build -b esp32
-- west build: generating a build system
Including boilerplate (Zephyr base): /projects/github/ncs/zephyr/cmake/app/boilerplate.cmake
-- Application: /projects/github/ncs/zephyr/samples/hello_world
-- Zephyr version: 2.2.99 (/projects/github/ncs/zephyr)
-- Found Python3: /usr/bin/python3.6 (found suitable exact version "3.6.9") found components: Interpreter 
-- Board: esp32
-- Found west: /home/tora/.local/bin/west (found suitable version "0.7.2", minimum required is "0.7.1")
-- Found dtc: /opt/zephyr-sdk/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found toolchain: zephyr (/opt/zephyr-sdk)
-- Found BOARD.dts: /projects/github/ncs/zephyr/boards/xtensa/esp32/esp32.dts
-- Generated zephyr.dts: /projects/github/ncs/zephyr/samples/hello_world/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /projects/github/ncs/zephyr/samples/hello_world/build/zephyr/include/generated/devicetree_unfixed.h
Parsing /projects/github/ncs/zephyr/Kconfig
Loaded configuration '/projects/github/ncs/zephyr/boards/xtensa/esp32/esp32_defconfig'
Merged configuration '/projects/github/ncs/zephyr/samples/hello_world/prj.conf'
Configuration saved to '/projects/github/ncs/zephyr/samples/hello_world/build/zephyr/.config'
Kconfig header saved to '/projects/github/ncs/zephyr/samples/hello_world/build/zephyr/include/generated/autoconf.h'
CMake Error at /projects/github/ncs/zephyr/cmake/extensions.cmake:1479 (message):
  No such file or directory: LIBGCC_FILE_NAME: ''
Call Stack (most recent call first):
  /projects/github/ncs/zephyr/cmake/compiler/gcc/target.cmake:65 (assert_exists)
  /projects/github/ncs/zephyr/cmake/target_toolchain.cmake:49 (include)
  /projects/github/ncs/zephyr/cmake/app/boilerplate.cmake:516 (include)
  /projects/github/ncs/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /projects/github/ncs/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:35 (include_boilerplate)
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /usr/bin/cmake -B/projects/github/ncs/zephyr/samples/hello_world/build -S/projects/github/ncs/zephyr/samples/hello_world -GNinja -DBOARD=Parsing /projects/github/ncs/zephyr/Kconfig
Loaded configuration '/projects/github/ncs/zephyr/boards/xtensa/esp32/esp32_defconfig'
Merged configuration '/projects/github/ncs/zephyr/samples/hello_world/prj.conf'
Configuration saved to '/projects/github/ncs/zephyr/samples/hello_world/build/zephyr/.config'
Kconfig header saved to '/projects/github/ncs/zephyr/samples/hello_world/build/zephyr/include/generated/autoconf.h'
CMake Error at /projects/github/ncs/zephyr/cmake/extensions.cmake:1479 (message):
  No such file or directory: LIBGCC_FILE_NAME: ''
Call Stack (most recent call first):
  /projects/github/ncs/zephyr/cmake/compiler/gcc/target.cmake:65 (assert_exists)
  /projects/github/ncs/zephyr/cmake/target_toolchain.cmake:49 (include)
  /projects/github/ncs/zephyr/cmake/app/boilerplate.cmake:516 (include)
  /projects/github/ncs/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /projects/github/ncs/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:35 (include_boilerplate)
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /usr/bin/cmake -B/projects/github/ncs/zephyr/samples/hello_world/build -S/projects/github/ncs/zephyr/samples/hello_world -GNinja -DBOARD=esp32
```
@pabigot can you confirm that this is actually still happening on earlier commits ?

@pabigot  @nashif  To me, it appears that this happened with Zephyr SDK 0.11.1 in this commit:
https://github.com/zephyrproject-rtos/zephyr/commit/5267a5e80976086be0b2188d0f0e3b49b53ae6e1

```
$ git checkout 5267a5e80976086be0b2188d0f0e3b49b53ae6e1
Previous HEAD position was 47ec94a662 ci: Bump docker image and SDK support to 0.11.1
HEAD is now at 5267a5e809 toolchain: Bump min SDK version to 0.11.1
$ rm -rf build; west build -b esp32

 -- west build: build configuration:
       source directory: /projects/github/ncs/zephyr/samples/hello_world
       build directory: /projects/github/ncs/zephyr/samples/hello_world/build (created)
       BOARD: esp32 (origin: command line)
-- west build: generating a build system
-- Zephyr version: 2.1.99
-- Found PythonInterp: /usr/bin/python3.6 (found suitable version "3.6.9", minimum required is "3.6") 
-- Selected BOARD esp32
-- Found west: /home/tora/.local/bin/west (found suitable version "0.7.2", minimum required is "0.6.0")
-- Loading /projects/github/ncs/zephyr/boards/xtensa/esp32/esp32.dts as base
Devicetree header saved to '/projects/github/ncs/zephyr/samples/hello_world/build/zephyr/include/generated/devicetree_unfixed.h'
Parsing /projects/github/ncs/zephyr/Kconfig
Loaded configuration '/projects/github/ncs/zephyr/boards/xtensa/esp32/esp32_defconfig'
Merged configuration '/projects/github/ncs/zephyr/samples/hello_world/prj.conf'
Configuration saved to '/projects/github/ncs/zephyr/samples/hello_world/build/zephyr/.config'
Kconfig header saved to '/projects/github/ncs/zephyr/samples/hello_world/build/zephyr/include/generated/autoconf.h'

warning: ENTROPY_ESP32_RNG (defined at drivers/entropy/Kconfig.esp32:6) was assigned the value 'y'
but got the value 'n'. Check these unsatisfied dependencies: ENTROPY_GENERATOR (=n). See
http://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_ENTROPY_ESP32_RNG.html and/or look up
ENTROPY_ESP32_RNG in the menuconfig/guiconfig interface. The Application Development Primer, Setting
Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful
too.

CMake Error at /projects/github/ncs/zephyr/cmake/extensions.cmake:1479 (message):
  No such file or directory: LIBGCC_FILE_NAME: ''
Call Stack (most recent call first):
  /projects/github/ncs/zephyr/cmake/compiler/gcc/target.cmake:69 (assert_exists)
  /projects/github/ncs/zephyr/cmake/target_toolchain.cmake:49 (include)
  /projects/github/ncs/zephyr/cmake/app/boilerplate.cmake:473 (include)
  CMakeLists.txt:5 (include)


-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /usr/bin/cmake -B/projects/github/ncs/zephyr/samples/hello_world/build -S/projects/github/ncs/zephyr/samples/hello_world -GNinja -DBOARD=esp32
$ 
```

--- a/cmake/toolchain/zephyr/host-tools.cmake
+++ b/cmake/toolchain/zephyr/host-tools.cmake
@@ -39,7 +39,16 @@ if(("zephyr" STREQUAL ${ZEPHYR_TOOLCHAIN_VARIANT}) OR
   SET(CMAKE_FIND_PACKAGE_SORT_ORDER NATURAL)
 
   if(DEFINED ZEPHYR_SDK_INSTALL_DIR)
+    # The Zephyr SDK will automatically set the toolchain variant.
+    # To support Zephyr SDK tools (DTC, and other tools) with 3rd party toolchains
+    # then we keep track of current toolchain variant.
+    set(ZEPHYR_CURRENT_TOOLCHAIN_VARIANT ${ZEPHYR_TOOLCHAIN_VARIANT})
     find_package(Zephyr-sdk ${MINIMUM_REQUIRED_SDK_VERSION} QUIET HINTS $ENV{ZEPHYR_SDK_INSTALL_DIR})
+    if(ZEPHYR_CURRENT_TOOLCHAIN_VARIANT)
+      if(NOT "zephyr" STREQUAL ${ZEPHYR_CURRENT_TOOLCHAIN_VARIANT})
+        set(ZEPHYR_TOOLCHAIN_VARIANT ${ZEPHYR_CURRENT_TOOLCHAIN_VARIANT})
+      endif()
+    endif()
   else()
     find_package(Zephyr-sdk ${MINIMUM_REQUIRED_SDK_VERSION} QUIET PATHS
                  /usr

